testnet/stacks-node/src/nakamoto_node/relayer.rs

```diff
@@ -856,65 +1059,102 @@ impl RelayerThread {
+            .map(|cmt| cmt.get_burn_view_consensus_hash() != &sort_tip.consensus_hash)
+            .unwrap_or(true);
+
+        // did our view of the highest ongoing tenure change?
+        // e.g. did we process blocks in an ancestral tenure that previously was empty?
+        // if so, then we will need to send (more likely, RBF) a block-commit to confirm it.
+        let (ongoing_tenure_consensus_hash, tenure_epoch_id) = if let Some(ongoing_tenure) =
+            NakamotoChainState::get_ongoing_tenure(&mut self.chainstate.index_conn(), &stacks_tip)
+                .map_err(|e| {
+                    error!(
+                        "Failed to get ongoing tenure off of {}: {:?}",
+                        &stacks_tip, &e
+                    );
+                    e
+                })
+                .ok()
+                .flatten()
+        {
+            (
+                ongoing_tenure.tenure_id_consensus_hash,
+                StacksEpochId::Epoch30,
+            )
+        } else if let Some(header) =
+            StacksChainState::get_stacks_block_header_info_by_index_block_hash(
                 self.chainstate.db(),
-                &chain_tip_header.consensus_hash,
+                &stacks_tip,
             )
-        else {
-            warn!("Failure getting the first block of tenure in order to assemble block commit";
-                  "tenure_consensus_hash" => %chain_tip_header.consensus_hash,
-                  "tip_block_hash" => %chain_tip_header.anchored_header.block_hash());
+            .map_err(|e| {
+                error!(
+                    "Failed to get stacks 2.x block header for {}: {:?}",
+                    &stacks_tip, &e
+                );
+                e
+            })
+            .ok()
+            .flatten()
+        {
+            (header.consensus_hash, StacksEpochId::Epoch25)
+        } else {
+            error!("Could not deduce ongoing tenure");
             return None;
         };
 
-        let chain_tip_tenure_id = chain_tip_tenure_start.index_block_hash();
-        let should_commit = burnchain_changed
-            || if let Some((_, last_committed_tenure_id)) = self.last_committed.as_ref() {
-                // if the tenure ID of the chain tip has changed, issue a new commit
-                last_committed_tenure_id != &chain_tip_tenure_id
-            } else {
-                // should be unreachable, but either way, if
-                //  `self.last_committed` is None, we should issue a commit
-                true
-            };
-
-        if should_commit {
-            Some(RelayerDirective::IssueBlockCommit(
-                chain_tip_tenure_start.consensus_hash,
-                chain_tip_tenure_start.anchored_header.block_hash(),
-            ))
-        } else {
-            None
+        let highest_tenure_changed = self
+            .last_committed
+            .as_ref()
+            .map(|cmt| cmt.get_tenure_id() != &ongoing_tenure_consensus_hash)
```

Comment:
Is this check equivalent to `cmt.get_tenure_id() != stacks_tip_ch` ? If so, the loading of `ongoing_tenure...` can all be skipped.

Reply:
Yes, I believe so